### ISSUE TITLE
🚀[Feature]: Restore 5.1 compatibility, again

### ISSUE THREAD

**Issue by o-l-a-v** [PSModule/PSSemVer#29]
### Description

Without it triggering a major version bump, which removing 5.1 compatibility would be in my opinion, somewhere between [v1.0.1](https://github.com/PSModule/PSSemVer/releases/tag/v1.0.1) and current version, the `.psd1` file pushed to PowerShell Gallery now has `PowerShellVersion = '7.4'`, which makes Windows PowerShell unable to import the module.

Simply setting this back to `PowerShellVersion = '5.1'` makes the module seemingly work on Windows PowerShell.

<img width="569" alt="image" src="https://github.com/user-attachments/assets/567b4158-0af4-487e-9843-e4714a20d266" />

So, for the same reasons as in #3: Please restore 5.1 compatibility.

* Windows PowerShell needs SemVer the most, pwsh has `System.Management.Automation.SemanticVersion` / `[semver]` after all.
* I can't see anything in the published code to PowerShell Gallery which requires version 7.4.

---

One idea: Use PSScriptAnalyzer in the release pipeline to validate that it supports the versions you intend to target? Found this blog post:

* <https://devblogs.microsoft.com/powershell/using-psscriptanalyzer-to-check-powershell-version-compatibility/>

Tested with following, which gave no errors:

```pwsh
Invoke-ScriptAnalyzer -Path C:\Users\olavb\Desktop\Ttes\PSSemVer\PSSemVer.psm1 -Settings @{
     Rules = @{
         PSUseCompatibleSyntax = @{
             # This turns the rule on (setting it to false will turn it off)
             Enable = $true
 
             # List the targeted versions of PowerShell here
             TargetVersions = @('5.1','7.4')
         }
     }
}
```

**Comment by MariusStorhaug** [comment]
Looking at it!

**Comment by MariusStorhaug** [comment]
I did a direction change in PSModule when doubling down on the `GitHub` module, to only focus on supporting LTS version of PowerShell, which I am still holding to. But with a caveat to allow supportability with 5.1 where little effort is required. I am however now building in testing for 5.1 for this reason. But here we go, I added an override for PowerShellVersion in the manifest so it will register as a 5.1 compatible module, but no testing to support this is added.

[https://github.com/PSModule](https://github.com/PSModule#supported-platforms#supported-platforms)

The break was with v1.1.0 of PSSemVer, and it should likely have been a major (2.0.0) change instead. I am trying to get better on implementing semver in practice (irony at its best :) )

Will see if i can add a simple "addon" test for 5.1 that doesn't interfere with the normal Process-PSModule workflow.
